src/components/Api.js

```diff
@@ -0,0 +1,89 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  //profile and avatar
+  async getProfile() {
+    return fetch(this._baseUrl + "/users/me/", {
+      method: "GET",
+      headers: this._headers,
+    }).then(this._checkResponse);
+  }
+
+  async patchProfileAvatar(newLink) {
+    return fetch(this._baseUrl + "/users/me/avatar", {
+      method: "PATCH",
+      headers: {
+        ...this._headers,
+        "Content-Type": "application/json",
```

Comment:
`        "Content-Type": "application/json",
`

* According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)



```js

      headers: this._headers,

```